src/main/kotlin/com/hj/leetcode/kotlin/problem731/Solution.kt

```diff
@@ -0,0 +1,54 @@
+package com.hj.leetcode.kotlin.problem731
+
+/**
+ * LeetCode page: [731. My Calendar II](https://leetcode.com/problems/my-calendar-ii/);
+ */
+class MyCalendarTwo {
+    private class Slot(
+        val start: Int,
+        val end: Int,
+        var next: Slot?,
+    )
+
+    private val headSlot = Slot(0, 0, null) // Sorted by start
+
+    /* Complexity for N calls:
+     * Time O(N^2) and Space O(N).
+     */
+    fun book(
+        start: Int,
+        end: Int,
+    ): Boolean {
+        val new = Slot(start, end, null)
+        val pendingEnds = mutableListOf<Int>()
+
+        // Find the position to insert the new slot
+        var insertionPoint = headSlot
+        while (true) {
+            pendingEnds.removeIf { it <= insertionPoint.start }
+            pendingEnds.add(insertionPoint.end)
+
+            val next = insertionPoint.next ?: break
+            if (new.start < next.start) {
+                break
+            }
+            insertionPoint = next
+        }
+
+        // Check is there any triple booking if we do the insertion
+        new.next = insertionPoint.next
+        var currentSlot: Slot? = new
+        while (currentSlot != null && currentSlot.start <= new.end) {
```

Comment:
We don't need to consider the case where `currentSlot.start = new.end`

Reply:
Addressed in https://github.com/hj-core/hj-leetcode-kotlin/pull/474/commits/f6720d6823d846dbf738102d4811f79d70184b3d